api/models/BedEvent.js

```diff
@@ -0,0 +1,92 @@
+/* global BedEvent */
+'use strict';
+
+var LinkingModels = require('sails-linking-models');
+var ModelHelpers = require('../lib/ModelHelpers');
+
+const operations = {
+  OPERATION_OUT_OF_COMMISSION: 'out commission',
+  OPERATION_IN_COMMISSION: 'in commission'
+};
+
+const reasons = {
+  REASON_SINGLE_OCCUPANCY: "Single Occupancy",
+  REASON_DAMAGE: "Maintenance - Malicious/Accidental Damage",
+  REASON_HEALTH_AND_SAFETY: "Maintenance - Health and Safety Concern",
+  REASON_PLANNED: "Maintenance – Planned works",
+  REASON_CRIME: "Crime Scene",
+  REASON_MEDICAL: "Medical Isolation",
+  REASON_OTHER: "Other"
+};
+
+const model = {
+  autoCreatedAt: true,
+  autoUpdatedAt: true,
+  reasons: reasons,
+  schema: true,
+  attributes: {
+    bed: {
+      model: "bed",
+      required: true,
+      defaultsTo: null
+    },
+    detainee: {
+      model: "detainee",
+      required: false
+    },
+    timestamp: {
+      type: 'datetime',
+      required: true
+    },
+    operation: {
+      type: 'string',
+      required: true,
+      enum: Object.keys(operations).map((k) => operations[k])
+    },
+    reason: {
+      type: 'string',
+      required: false,
+      defaultsTo: null,
+      enum: Object.keys(reasons).map((k) => reasons[k])
+    },
+    active: {
+      type: 'boolean',
+      required: true
+    }
+  },
+
+  getOOCByCentreGroupByGenderAndReason: (cid) =>
```

Comment:
`cid` is quite a loaded term in this domain, please rename it to something like `centre_id` which we use elsewhere iirc